python/dgl/nn/pytorch/hetero.py

```diff
@@ -381,9 +374,9 @@ def forward(self, hg):
         embed : Tensor
             Node embeddings.
         """
-        embed = th.tensor([])
+        nodes = hg.num_nodes()
         g = to_homogeneous(hg)
-        for ntype in self.embed.keys():
-            embed = th.cat((embed, self.embed[ntype]))
+        embed = nn.Parameter(th.FloatTensor(nodes, self.embed_size))
+        nn.init.xavier_uniform_(embed, gain=nn.init.calculate_gain('relu'))
```

Comment:
Previously, you treat `embed` as an attribute of the class. This allows users to optimize it by passing HeteroEmbedding(...).parameters() to an optimizer. Otherwise, users need to handle that with the returned embed. I still prefer the previous approach as the latest one can be error-prone.

Reply:
OK, I did not consider this, and I will put them in the __init__ of the class.